node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -335,23 +329,53 @@ object WebApi {
   private def exprFromBundleProto(b: Bundle): Option[RhoExpr] = exprFromParProto(b.body)
 
   private def mkUnforgExpr(f: String => RhoUnforg, bs: ByteString): ExprUnforg =
-    ExprUnforg(f(toHex(bs)))
+    ExprUnforg(f(bs.toHexString))
 
   // RhoExpr to protobuf
 
+  private def rhoExprToParProto(exp: RhoExpr): Par = {
+    def exprProtoToParProto(exp: Expr): Par = Par(exprs = Seq(exp))
+
+    exp match {
+      // Nested expressions (Par, Tuple, List and Set are converted to JSON list)
+      case ExprPar(data) =>
+        exprProtoToParProto(Expr().withEListBody(EList(data.map(rhoExprToParProto))))
+      case ExprTuple(data) =>
+        exprProtoToParProto(Expr().withETupleBody(ETuple(data.map(rhoExprToParProto))))
+      case ExprList(data) =>
+        exprProtoToParProto(Expr().withEListBody(EList(data.map(rhoExprToParProto))))
+      case ExprSet(data) =>
+        exprProtoToParProto(Expr().withESetBody(ParSet(data.map(rhoExprToParProto))))
+      case ExprMap(data) =>
+        exprProtoToParProto(Expr().withEMapBody(ParMap(data.map {
+          case (k, v) => (rhoExprToParProto(ExprString(k)), rhoExprToParProto(v))
+        }.toList)))
+      // Terminal expressions (here is the data)
+      case ExprBool(data)   => exprProtoToParProto(Expr().withGBool(data))
```

Comment:
This can then be `Boolean(data)`.

Reply:
Can you explain please? Do you mean this change?
```scala
case Boolean(data) => Expression(Expr().withGBool(data))
```
Why only for `Boolean`? What about `Long` and `String`? Why it is necessary?

Reply:
Oh, I see what you mean
https://github.com/rchain/rchain/blob/0bfc2aa13e73a27c1f21200dee7336bd252ea17a/rholang/src/main/scala/coop/rchain/rholang/interpreter/RhoType.scala#L42-L50